uni_client.go

```diff
@@ -63,14 +61,22 @@ func DialUniWithContext(ctx context.Context, lggr Logger, target string, ed25519
 		return nil, err
 	}
 
-	conn, err := retryConnectWithBackoff(ctx, lggr, func(ctx2 context.Context) (Conn, error) {
-		return connect(ctx2, target, tlsConfig)
-	})
-	if err != nil {
-		return nil, err
+	u := NewTLSUniClientConn(lggr, target, tlsConfig)
+	return u, u.Dial(ctx)
+}
+
+func NewTLSUniClientConn(lggr Logger, target string, tlsConfig *tls.Config) *UniClientConn {
```

Comment:
Alternatively, we could have a more general: 
```suggestion
func NewUniClientConn(lggr Logger, connectFn func(context.Context) (Conn, error)) *UniClientConn {
```
With:
```go
func NewTLSConnFn(target string, *tls.Config) func(context.Context) (Conn, error))
```
So callers do:
```go
conn := wsrpc.NewUniClientConn(lggr, wsrpc.NewTLSConnFn(target, tlsConfig)) 
```

Reply:
I think more specific is best given what WSRPC was built to handle

Reply:
`DialUniWithContext` was limited to a `ed25519PrivKey ed25519.PrivateKey`. We could create some variant that supports `crypto.Signer`, but it's trivial to just let the callers construct the `tls.Config` themselves instead, which avoids choosing a least-bad name for a new `DialUniWithContext` variant.